artemis/module_base.py

```diff
@@ -559,6 +606,17 @@ def process_multiple(self, tasks: List[Task]) -> None:
         if len(tasks) == 0:
             return
 
+        # Load configuration from first task's payload if present
```

Comment:
IMO if there are multiple tasks with separate configurations, we should process them separately, not just take the config for the first one and call it a day ;)